docker-app/qfieldcloud/core/rest_utils.py

```diff
@@ -1,14 +1,22 @@
 import logging
+import sys
 
 from django.core import exceptions
 from qfieldcloud.core import exceptions as qfieldcloud_exceptions
+from qfieldcloud.core.exceptions import QFieldCloudException
 from rest_framework import exceptions as rest_exceptions
 from rest_framework.response import Response
 
 logger = logging.getLogger(__name__)
 
 
 def exception_handler(exc, context):
+
+    # When running tests, we don't handle unexpected exceptions, as it swallows the
+    # stack trace, making debugging much more difficult
+    if sys.argv[1] == "test" and not isinstance(exc, QFieldCloudException):
```

Comment:
Is there a better way to get we are withing a test? E.g. `ENVIRONMENT=test` envvar or something? This might fail if the `test` command is renamed for whatever reason.

Not a blocker to merge, just looking for a cleaner solution.

Reply:
Yes I agree, but from what I found there's no out of the box way to detect that. I saw people suggesting to subclass django.test.runner.DiscoverRunner to add such a variable ?

Alternatively it we could use settings.DEBUG (which is more or less how django behaves with normal frontend view: if debug=False it shows the 500 page, if debug=True it shows the debug information).

But that would mean that the clients would not get a proper error json error message if debug=True. We would have to run with debug=False on deployment (including staging), but that's anyway a good idea IMO.

Reply:
yup, I got to the same conclusion a while ago.

The problem I have is that `DEBUG=True` is not directly linked to the tests. E.g. I can imagine one might want to run it on staging temporary to extract debug information, which might also alter the behaviour of some of the modules. 

Now I checked - we run CI tests with `DEBUG=1` on the CI, which I would propose to change, IMO tests with debug enabled should be run only locally for ... debugging.

Therefore I think subclass `django.test.runner.DiscoverRunner` is indeed probably the cleanest solution, which will improve our quality of life, unless I am misisng some extra complexity from it?

Reply:
I don't think there's any extra complexity, will make these changes